File: src/0035.search-insert-position.35/solution.ts
function searchInsert (nums: number[], target: number): number {
    let result = nums.length - 1;
    let low = 0;
    let high = nums.length;
    while (low <= high) {
        const mid = (low + high) >> 1;
        if (nums[mid] < target) {
            low = mid + 1;
        } else if (nums[mid] > target) {
            high = mid - 1;
            result = Math.min(result, mid);
        } else {
            return mid;
        }
    }
    return result;
}

File: src/0035.search-insert-position.35/solution.java
class Solution {
    public int searchInsert(int[] nums, int target) {
        int low = 0;
        int high = nums.length-1;
        int result = nums.length;
        while (low<=high){
            int mid = (low+high)>>1;
            if(nums[mid]<target){
                low = mid+1;
            }else{
                result = mid;
                high = mid-1;
            }
        }
        return result;
    }
}